src/components/sbb-file-selector/sbb-file-selector.tsx

```diff
@@ -0,0 +1,289 @@
+
+  public connectedCallback(): void {
+    this._handlerRepository.connect();
+  }
+
+  public disconnectedCallback(): void {
+    this._handlerRepository.disconnect();
+  }
+
+  private _blockEvent(event): void {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
+  private _checkFileEquality(file1: File, file2: File): boolean {
+    return (
+      file1.name === file2.name &&
+      file1.size === file2.size &&
+      file1.lastModified === file2.lastModified
+    );
+  }
+
+  private _openFileWindow(): void {
+    if (!this.disabled) {
+      return this._hiddenInput.click();
+    }
+  }
+
+  private _onDragEnter(event): void {
+    if (!this.disabled) {
+      this._setDragState(event.target, true);
+      this._blockEvent(event);
+    }
+  }
+
+  private _onDragLeave(event): void {
+    if (!this.disabled && event.target === this._dragTarget) {
+      this._setDragState();
+      this._blockEvent(event);
+    }
+  }
+
+  private _onFileDrop(event): void {
+    if (!this.disabled) {
+      this._setDragState();
+      this._blockEvent(event);
+      try {
+        const files = this.multiple ? event.dataTransfer.files : [event.dataTransfer.files[0]];
+        this._createFilesList(files);
+      } catch (e) {
+        this.error.emit(e);
+      }
+    }
+  }
+
+  private _setDragState(dragTarget = undefined, isDragEnter = false): void {
+    if (this.variant === 'dropzone') {
+      this._dragTarget = dragTarget;
+      toggleDatasetEntry(this._element, 'active', isDragEnter);
+      // FIXME the next line causes button flickering; same if directly setting the button background-color variable
+      toggleDatasetEntry(this._loadButton, 'active', isDragEnter);
+    }
+  }
+
+  private _readFiles(event): void {
+    try {
+      if (event.target.files) {
+        this._createFilesList(event.target.files);
+      }
+    } catch (e) {
+      this.error.emit(e);
+    }
+  }
+
+  private _createFilesList(files: File[]): void {
+    if (
+      !this.multiple ||
+      this.multipleMode !== 'persistent' ||
+      !this.files ||
+      this.files.length === 0
+    ) {
+      this.files = Array.from(files);
+    } else {
+      this.files = Array.from(files)
+        .filter(
+          (newFile: File) =>
+            this.files.findIndex((oldFile: File) => this._checkFileEquality(newFile, oldFile)) ===
+            -1,
+        )
+        .concat(this.files);
+    }
+    this.fileChangedEvent.emit(this.files);
+  }
+
+  private _removeFile(file: File): void {
+    this.files = this.files.filter((f: File) => !this._checkFileEquality(file, f));
+    this.fileChangedEvent.emit(this.files);
+  }
+
+  /** Calculates the correct unit for the file's size. */
+  private _formatFileSize(size: number): string {
+    const i: number = Math.floor(Math.log(size) / Math.log(1024));
+    return `${(size / Math.pow(1024, i)).toFixed(0)} ${this._suffixes[i]}`;
+  }
+
+  private _renderDefaultMode(): JSX.Element {
+    return (
+      <sbb-button
```

Comment:
This might have been a misunderstanding. The button was supposed to be `is-static` and the label should be wrapped around it.
We did not explicitely state this, which is on us.
Do you think it is possible to refactor this?
Look at the code of https://angular.app.sbb.ch/angular/components/file-selector/examples for a reference.

Reply:
Done in [b757080](https://github.com/lyne-design-system/lyne-components/pull/1894/commits/b757080eb29d0c5d03c4a0e5edf5e769d2832fc4), it could be good?